pkg/apis/pingcap/v1alpha1/validation/validation.go

```diff
@@ -146,65 +146,70 @@ func validateTiCDCSpec(spec *v1alpha1.TiCDCSpec, fldPath *field.Path) field.Erro
 	return allErrs
 }
 
-func validateTiFlashConfig(config *v1alpha1.TiFlashConfig, path *field.Path) field.ErrorList {
+func validateTiFlashConfig(config *v1alpha1.TiFlashConfigWraper, path *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if config == nil {
 		return allErrs
 	}
 
-	if config.CommonConfig != nil {
-		if config.CommonConfig.Flash != nil {
-			if config.CommonConfig.Flash.OverlapThreshold != nil {
-				if *config.CommonConfig.Flash.OverlapThreshold < 0 || *config.CommonConfig.Flash.OverlapThreshold > 1 {
+	if config.Common != nil {
+		if v := config.Common.Get("flash.overlap_threshold"); v != nil {
+			if value, err := v.AsFloat(); err == nil {
+				if value < 0 || value > 1 {
 					allErrs = append(allErrs, field.Invalid(path.Child("config.config.flash.overlap_threshold"),
-						config.CommonConfig.Flash.OverlapThreshold,
+						value,
 						"overlap_threshold must be in the range of [0,1]."))
 				}
 			}
```

Comment:
Append errors for the `err!=nil` case? same for the following validation cases.

Reply:
https://github.com/pingcap/tidb-operator/pull/3355/commits/ae214056f7f2ddcea2303f02eecadfc83e2bfbbd